### ISSUE TITLE
Class constructors missing in `pcui.d.ts`

### ISSUE THREAD

**Issue by kungfooman** [playcanvas/pcui#115]
Currently only `@class` tagged functions have constructors in `pcui.d.ts`, because JSDoc demands `@param` for the constructor before the class itself and not before the actual constructor.

E.g. as it is currently:

```js
/**
 * @name Container
 * @class
 * @classdesc A container is the basic building block for Elements that are grouped together.
 * A container can contain any other element including other containers.
 * @augments Element
 * @mixes IContainer
 * @mixes IFlex
 * @mixes IGrid
 * @mixes IScrollable
 * @mixes IResizable
 */
class Container extends Element {
    /**
     * Creates a new Container.
     *
     * @param {object} args - The arguments. Extends the pcui.Element constructor arguments. All settable properties can also be set through the constructor.
     * @param {HTMLElement} [args.dom] - The DOM element to use for the container. If unspecified a new element will be created.
     */
    constructor(args) {
```

Would require a rewrite into:

```js
/**
 * @name Container
 * @class
 * @classdesc A container is the basic building block for Elements that are grouped together.
 * A container can contain any other element including other containers.
 * @augments Element
 * @mixes IContainer
 * @mixes IFlex
 * @mixes IGrid
 * @mixes IScrollable
 * @mixes IResizable
 * @param {object} args - The arguments. Extends the pcui.Element constructor arguments. All settable properties can also be set through the constructor.
 * @param {HTMLElement} [args.dom] - The DOM element to use for the container. If unspecified a new element will be created.
 */
class Container extends Element {
    constructor(args) {
```

And only then the `constructor` pops up in `pcui.d.ts`:

![image](https://user-images.githubusercontent.com/5236548/121772536-30c92480-cb76-11eb-8fb9-455ac93fd318.png)

IMO JSDoc should support both styles (first JSDoc style is using in `playcanvas` itself), does anyone happen to know why JSDoc is so picky with this? VSCode/TypeScript e.g. can't even pick up the JSDoc style... so this just creates incompatibilities between toolings.

**Comment by illeatmyhat** [comment]
Things like this make the PCUI tools, observer, graph, essentially unusable in TypeScript.
Even just importing things is way too difficult.

**Comment by kungfooman** [comment]
> Things like this make the PCUI tools, observer, graph, essentially unusable in TypeScript.
> Even just importing things is way too difficult.

Well, I feel your frustration, I made a PR [1] about this but not even `cross-env` gets merged for Windows compatibility. You can review my PR and add your two cents. Not sure how many months to wait for "internal discussions".

I wouldn't get my hopes up too much tho, the code seems deadlocked by JSDoc peculiarities entrenched by the documentation generator.

[1] https://github.com/playcanvas/pcui/pull/119

**Comment by willeastcott** [comment]
Fixed by #190.
